media_utils/video_puppeteer.py

```diff
@@ -187,12 +198,18 @@ def playback_done(video):
     # either still playing or there is a problem, so this will never
     # return True.
 
+    active_re = re.compile('NextGetSampleIndex:([\d]+)')
+
     debug_lines = video.get_debug_lines()
     if debug_lines is not None:
+        for occurence in active_re.findall(' '.join(debug_lines)):
+            if occurence.group(1) == '0':
+                raise VideoException('Video %s is stalled' % video)
```

Comment:
My understanding is that __if__ there is a stall (no progress is being made) then NextGetSampleIndex==0 is a problem. "This would
indicate an internal failure (but only if there's an actual stall)"

jya also describes a way to detect a stall based on buffered ranges, but I think that should be developed in a separate method (and as part of a different bug). (e.g. there is a `player_stalled` method in `youtube_puppeteer.py`